sycl/CMakeLists.txt

```diff
@@ -180,6 +180,8 @@ configure_file("${version_header}.in" "${version_header}")
 # Copy SYCL headers
 add_custom_target(sycl-headers ALL
 COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/CL ${SYCL_INCLUDE_BUILD_DIR}/CL
+COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/sycl ${SYCL_INCLUDE_BUILD_DIR}/sycl
```

Comment:
I think `sycl` directory name may conflict with [backend generalization proposal](https://github.com/KhronosGroup/SYCL-Shared/blob/master/proposals/sycl_generalization.md#the-sycl-programming-model) on Windows since filenames are case insensitive by default.

Reply:
@kbobrovs ping

Reply:
@alexbatashev, can you elaborate a bit more what problem with directory name do you see?
BTW, we might need to move the library into  sycl/intel/gpu to reflect namespace. @bader - is this a requirement or we can leave SIMD library implementation in sycl/esimd?
Also, we don't use __SYCL_INLINE_NAMESPACE(cl) for the esimd sources anticipating dropping the cl:: in the future SYCL spec. Is that OK?

Reply:
@kbobrovs am I right that one would need to include this headers as `#include <sycl/esimd/...>`? If yes, on Linux we'd have `#include <SYCL/...>` and `#include <sycl/...>` which seems a bit confusing.

Reply:
@alexbatashev , still don't get it, sorry. Why #include <SYCL/...> on Linux?

Reply:
OK, @DenisBakhvalov clarified that it may be related to 'SYCL' folder name (in capital) in the new SYCL 2020. If that's the case, we can move ESIMD stuff under 'SYCL' from 'sycl' when this happens, no problem.